yt_dlp/extractor/tbsjp.py

```diff
@@ -0,0 +1,161 @@
+        episode = traverse_obj(meta, ('falcorCache', 'catalog', 'episode', video_id, 'value'))
+
+        episode_meta = traverse_obj(episode, {
+            'categories': 'keywords',
+            'id': 'content_id',
+            'description': ('description', 0, 'value'),
+            'timestamp': ('created_at', {unified_timestamp}),
+            'release_timestamp': ('pub_date', {unified_timestamp}),
+            'duration': ('tv_episode_info', 'duration'),
+            'episode_number': ('tv_episode_info', 'episode_number'),
+            'series': ('custom_data', 'program_name'),
+        })
+
+        tf_url = urljoin(
+            url,
+            self._search_regex(r'<script src=["\'](/assets/tf\.[^"\']+\.js)["\']', page, 'stream API config')
+        )
+        tf_js = self._download_webpage(tf_url, video_id, note='Downloading stream API config')
+        video_url = self._search_regex(r'videoPlaybackUrl: *[\'"]([^\'"]+)[\'"]', tf_js, 'stream API url')
+        api_key = self._search_regex(r'api_key: *[\'"]([^\'"]+)[\'"]', tf_js, 'stream API key')
+
+        try:
+            source_meta = self._download_json(f'{video_url}ref:{video_id}', video_id,
+                                              headers={'X-Streaks-Api-Key': api_key},
+                                              note='Downloading stream metadata')
+        except ExtractorError as e:
+            if isinstance(e.cause, HTTPError) and e.cause.status == 403:
+                self.raise_geo_restricted(countries=['JP'])
+            raise
+
+        formats = []
+        subs = []
+        for i in traverse_obj(source_meta, ('sources', ..., 'src')):
+            format, sub = self._extract_m3u8_formats_and_subtitles(i, video_id)
+            formats.extend(format)
+            subs.extend(sub)
+
+        return {
+            'title': get_element_text_and_html_by_tag('h3', page)[0],
+            'episode': traverse_obj(episode, ('title', lambda _, v: not v.get('is_phonetic'), 'value'), get_all=False),
+            **episode_meta,
+            'formats': formats,
+            'subtitles': sub,
+        }
+
+
+class TBSJPProgramIE(InfoExtractor):
+    _VALID_URL = r'https://cu\.tbs\.co\.jp/program/(?P<id>[\d]+)'
+    _TESTS = [{
+        'url': 'https://cu.tbs.co.jp/program/23601',
+        'playlist_mincount': 4,
+        'info_dict': {
+            'id': '23601',
+            'categories': ['エンタメ', 'ミライカプセル', '会社', '働く', 'バラエティ', '動画'],
+            'description': '幼少期の夢は大人になって、どう成長したのだろうか？\nそしてその夢は今後、どのように広がっていくのか？\nいま話題の会社で働く人の「夢の成長」を描く',
+            'series': 'ミライカプセル　-I have a dream-',
+            'title': 'ミライカプセル　-I have a dream-'
+        }
+    }]
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+        page = self._download_webpage(url, video_id)
+        meta = self._search_json('window.app=', page, 'programme info', video_id)
+
+        programme = traverse_obj(meta, ('falcorCache', 'catalog', 'program', video_id, 'false', 'value'))
```

Comment:
- same regex suggestion as above extractor
 - it's confusing to have this var be `video_id` since it's a playlist ID and never returned as the id of anything
```suggestion
        program_id = self._match_id(url)
        page = self._download_webpage(url, program_id)
        meta = self._search_json('window\.app\s*=', page, 'programme info', program_id)
        programme = traverse_obj(meta, ('falcorCache', 'catalog', 'program', program_id, 'false', 'value'))
```